pkg/controllers/state/cluster.go

```diff
@@ -181,6 +186,30 @@ func (c *Cluster) NominateNodeForPod(nodeName string) {
 	c.nominatedNodes.SetDefault(nodeName, nil)
 }
 
+// AddNominatedNodeEvictionObserver adds an observer function to be called when any cache entry from the
+// nominatedNodes cache expires
+func (c *Cluster) AddNominatedNodeEvictionObserver(f observerFunc) {
+	c.nominatedNodeObservers.Add(f)
+}
+
+// onNominatedNodeEviction is registered as the function called when a nominatedNode cache
+// entry expires. It will alert all registered observer functions by calling the registered function
+func (c *Cluster) onNominatedNodeEviction(key string, _ interface{}) {
+	if _, expiration, _ := c.nominatedNodes.GetWithExpiration(key); expiration.After(time.Now()) {
```

Comment:
This shouldn’t ever happen, right? If it’s been expired won’t the expiry always be before time.Now()?